packages/backend/src/services/vendor/baseSearchDriver.ts

```diff
@@ -1,22 +1,62 @@
-import * as superagent from 'superagent';
+import process from 'node:process';
+
+import superagent, {SuperAgentRequest} from 'superagent';
+import {HttpsProxyAgent} from 'https-proxy-agent';
 
 import {AbstractFields, VendorFieldType} from '../../../../shared/vendor/vendorModel';
 import {VendorResult} from '../../../../shared/api/response';
 import {VendorIntegration} from '../../../../shared/integration/vendorIntegration';
 import {Vendor} from '../../../../shared/vendor/vendor';
 import {DetailsOptions} from '../../models/detailsOptions';
+import {Logger, WithLogger} from '../logger';
 
 import {DetailsSearchDriver, SearchDriver} from './searchDriver';
 
+export class ProxyClient extends WithLogger {
+  private readonly client: typeof superagent;
+
+  constructor(private readonly env = process.env) {
+    super(new Logger());
+    this.client = superagent;
+  }
+
+  /**
+   * Uses a proxy if needed
+   */
+  protected request(url: URL, verb: 'get' | 'post' = 'get'): SuperAgentRequest {
+    let proxyUrl: string | undefined;
+
+    if (url.protocol === 'https:') {
+      proxyUrl = this.env.HTTPS_PROXY ?? this.env.https_proxy;
+      if (proxyUrl) {
+        if (!proxyUrl.startsWith('http:')) {
+          proxyUrl = `http://${proxyUrl}`;
```

Comment:
I have little contex, but what obout urls in the form //domain/...

Reply:
indeed, but unlikely, proxy urls in env vars are passed as: host:port or scheme:host:port